packages/graphql/tests/e2e/subscriptions/auth/authentication.int.test.ts

```diff
@@ -0,0 +1,583 @@
+
+            extend type ${typeMovie} @auth(rules: [{ isAuthenticated: false, operations: [SUBSCRIBE] }])
+            `;
+
+            const neoSchema = new Neo4jGraphQL({
+                typeDefs,
+                driver,
+                plugins: {
+                    subscriptions: new TestSubscriptionsPlugin(),
+                    auth: new Neo4jGraphQLAuthJWTPlugin({
+                        secret: "secret",
+                    }),
+                },
+            });
+
+            server = new ApolloTestServer(neoSchema);
+            await server.start();
+        });
+
+        afterEach(async () => {
+            await wsClient.close();
+        });
+
+        afterAll(async () => {
+            await server.close();
+        });
+
+        test("authenticated subscription fails", async () => {
+            wsClient = new WebSocketTestClient(server.wsPath, jwtToken);
+            await wsClient.subscribe(`
+                subscription {
+                    ${typeMovie.operations.subscribe.created} {
+                        ${typeMovie.operations.subscribe.payload.created} {
+                            title
+                        }
+                    }
+                }
+                `);
+
+            const result = await createMovie("movie1", server);
+
+            expect(result.body.errors).toBeUndefined();
+            expect(wsClient.events).toEqual([]);
+            expect(wsClient.errors).toEqual([expect.objectContaining({ message: "Error" })]);
+        });
+
+        test("unauthenticated subscription send events", async () => {
+            wsClient = new WebSocketTestClient(server.wsPath);
+            await wsClient.subscribe(`
+                    subscription {
+                        ${typeMovie.operations.subscribe.created} {
+                            ${typeMovie.operations.subscribe.payload.created} {
+                                title
+                            }
+                        }
+                    }
+                    `);
+
+            const result = await createMovie("movie1", server);
+
+            expect(result.body.errors).toBeUndefined();
+            expect(wsClient.events).toEqual([
+                {
+                    [typeMovie.operations.subscribe.created]: {
+                        [typeMovie.operations.subscribe.payload.created]: {
+                            title: "movie1",
+                        },
+                    },
+                },
+            ]);
+            expect(wsClient.errors).toEqual([]);
+        });
+    });
+
+    describe("auth with allowUnauthenticated", () => {
+        let server: TestGraphQLServer;
+        let wsClient: WebSocketTestClient;
+
+        beforeAll(async () => {
+            const typeDefs = `
+            type ${typeMovie} {
+                title: String!
+            }
+
+            extend type ${typeMovie} @auth(rules: [{ isAuthenticated: true, allowUnauthenticated: true, operations: [SUBSCRIBE] }])
```

Comment:
Perhaps out of the scope of this PR, but I feel like combining `isAuthenticated` and `allowUnauthenticated` shouldn't be allowed, and `allowUnauthenticated` should only be allowed to be combined with _authorization_ rules. We should discuss this!

Reply:
We need to discuss, isn't the point of allow unauthenticated to support whitelisting?